Makefile

```diff
@@ -25,26 +26,44 @@ all: test build package
 
 build: build-linux
 
-build-macos:
+build-darwin:
 	@echo "> Building for macos..."
-	mkdir -p ./out/lifecycle
-	GOOS=darwin $(GOENV) $(GOBUILD) -o ./out/lifecycle -a ./cmd/...
+	mkdir -p $(BUILD_DIR)/lifecycle
+	GOOS=darwin $(GOENV) $(GOBUILD) -o $(BUILD_DIR)/lifecycle -a ./cmd/launcher
+	GOOS=darwin $(GOENV) $(GOBUILD) -o $(BUILD_DIR)/lifecycle/detector -a ./cmd/build
+	ln -sf detector $(BUILD_DIR)/lifecycle/analyzer
+	ln -sf detector $(BUILD_DIR)/lifecycle/restorer
+	ln -sf detector $(BUILD_DIR)/lifecycle/builder
+	ln -sf detector $(BUILD_DIR)/lifecycle/exporter
+	ln -sf detector $(BUILD_DIR)/lifecycle/rebaser
 
 build-linux:
 	@echo "> Building for linux..."
-	mkdir -p ./out/lifecycle
-	GOOS=linux $(GOENV) $(GOBUILD) -o ./out/lifecycle -a ./cmd/...
+	mkdir -p $(BUILD_DIR)/lifecycle
+	GOOS=linux $(GOENV) $(GOBUILD) -o $(BUILD_DIR)/lifecycle -a ./cmd/launcher
+	GOOS=linux $(GOENV) $(GOBUILD) -o $(BUILD_DIR)/lifecycle/detector -a ./cmd/build
+	ln -sf detector $(BUILD_DIR)/lifecycle/analyzer
+	ln -sf detector $(BUILD_DIR)/lifecycle/restorer
+	ln -sf detector $(BUILD_DIR)/lifecycle/builder
+	ln -sf detector $(BUILD_DIR)/lifecycle/exporter
+	ln -sf detector $(BUILD_DIR)/lifecycle/rebaser
 
 build-windows:
 	@echo "> Building for windows..."
-	mkdir -p ./out/lifecycle
-	GOOS=windows $(GOENV) $(GOBUILD) -o ./out/lifecycle -a ./cmd/...
+	mkdir -p $(BUILD_DIR)/lifecycle
+	GOOS=linux $(GOENV) $(GOBUILD) -o $(BUILD_DIR)/lifecycle -a ./cmd/launcher
+	GOOS=linux $(GOENV) $(GOBUILD) -o $(BUILD_DIR)/lifecycle/detector -a ./cmd/build
```

Comment:
Couple things:

- `GOOS` should be windows.
- output file should end in .exe

Are we using this? Should we remove windows build until we verify that it works? AFAIK that support doesn't exist yet.